yb-voyager/src/srcdb/pg_dump_extract_schema.go

```diff
@@ -39,14 +42,23 @@ func pgdumpExtractSchema(source *Source, connectionUri string, exportDir string)
 		utils.ErrExit("could not get absolute path of pg_dump command: %v", err)
 	}
 
-	pgDumpArgs := fmt.Sprintf(`--schema-only --schema "%s" --no-owner -f %s --no-privileges --no-tablespaces --extension "*" --load-via-partition-root`,
-		source.Schema, filepath.Join(exportDir, "temp", "schema.sql"))
-	if !source.CommentsOnObjects {
-		pgDumpArgs = fmt.Sprintf(`%s --no-comments`, pgDumpArgs)
+	pgDumpArgs.Schema = source.Schema
+	pgDumpArgs.SchemaTempFilePath = filepath.Join(exportDir, "temp", "schema.sql")
+	pgDumpArgs.NoComments = strconv.FormatBool(!source.CommentsOnObjects)
+	pgDumpArgs.ExtensionPattern = `"*"`
+
+	args := source.getPgDumpSchemaArgsFromFile()
+	if args == "" {
+		args = fmt.Sprintf(`--schema-only --schema "%s" --no-owner -f %s --no-privileges --no-tablespaces --extension "*" --load-via-partition-root`,
+			source.Schema, filepath.Join(exportDir, "temp", "schema.sql"))
+		if !source.CommentsOnObjects {
```

Comment:
Why special handling for this arg? Won't it be read from the file anyway?

Reply:
yeah it is the case when the file is not present. I agree with you idea of having an internal pg-dump-args file also. So this will be gone then